extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -853,109 +932,110 @@ protected function  isCorporateWiki() {
-				'cat_entertainment' =>  isset( $filters['cat_entertainment'] ),
-				'cat_lifestyle' =>      isset( $filters['cat_lifestyle'] ),
-				'is_hd' =>              isset( $filters['is_hd'] ),
-				'is_image' =>           isset( $filters['is_image'] ),
-				'is_video' =>           isset( $filters['is_video'] ),
-				'sort_default' =>       $rank == 'default',
-				'sort_longest' =>       $rank == 'longest',
-				'sort_newest' =>        $rank == 'newest',
-				'no_filter' =>          !( isset( $filters['is_image'] ) || isset( $filters['is_video'] ) ),
-			);
+		$form = [
+			'by_category' => $this->getVal( 'by_category', false ),
+			'cat_videogames' => isset( $filters['cat_videogames'] ),
+			'cat_entertainment' => isset( $filters['cat_entertainment'] ),
+			'cat_lifestyle' => isset( $filters['cat_lifestyle'] ),
+			'is_hd' => isset( $filters['is_hd'] ),
+			'is_image' => isset( $filters['is_image'] ),
+			'is_video' => isset( $filters['is_video'] ),
+			'sort_default' => $rank == 'default',
+			'sort_longest' => $rank == 'longest',
+			'sort_newest' => $rank == 'newest',
+			'no_filter' => !( isset( $filters['is_image'] ) || isset( $filters['is_video'] ) ),
+		];
 
 		// Set video wiki to display only videos by default
-		if ( $is_video_wiki && $form['no_filter'] == 1 && !in_array( 'no_filter', $this->getVal( 'filters', array() ) ) ) {
+		if ( $is_video_wiki && $form['no_filter'] == 1 && !in_array( 'no_filter', $this->getVal( 'filters', [] ) ) ) {
 			$form['is_video'] = 1;
 			$form['no_filter'] = 0;
 		}
 
-		$this->setVal( 'bareterm', 			$config->getQuery()->getSanitizedQuery() );
-		$this->setVal( 'searchProfiles', 	$config->getSearchProfiles() );
-		$this->setVal( 'activeTab', 		$config->getActiveTab() );
-		$this->setVal( 'form',				$form );
-		$this->setVal( 'is_video_wiki',		$is_video_wiki );
+		$this->setVal( 'bareterm', $config->getQuery()->getSanitizedQuery() );
+		$this->setVal( 'searchProfiles', $config->getSearchProfiles() );
+		$this->setVal( 'activeTab', $config->getActiveTab() );
+		$this->setVal( 'form', $form );
+		$this->setVal( 'is_video_wiki', $is_video_wiki );
 	}
 
 	/**
 	 * This handles pagination via a template script.
+	 *
 	 * @see    SearchControllerTest::testPagination
 	 * @throws Exception
-	 * @return boolean|null (false if we don't want pagination, fully routed to view via sendSelfRequest if we do want pagination)
+	 * @return boolean|null (false if we don't want pagination, fully routed to view via sendSelfRequest if we do want
+	 *     pagination)
 	 */
 	public function pagination() {
-		$config = $this->getVal('config', false);
-		if ( ! $config || (! $config instanceOf Wikia\Search\Config ) ) {
-			throw new Exception("This should not be called outside of self-request context.");
+		$config = $this->getVal( 'config', false );
+		if ( !$config || ( !$config instanceOf Wikia\Search\Config ) ) {
+			throw new Exception( "This should not be called outside of self-request context." );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)